src/Surface.zig

```diff
@@ -3107,6 +3109,9 @@ fn mouseReport(
     mods: input.Mods,
     pos: apprt.CursorPos,
 ) !void {
+    // If mouse reporting is disabled, do nothing
+    if (!self.config.mouse_reporting) return;
```

Comment:
I don't think this is good enough or correct.

I'd change this to an assert and change all the upstream callers to check for this state.

To make things cleaner, I'd probably add a `fn isMouseReporting` that checks both the config and the terminal state. 

This is required so that text selection and so on as you said continues to work.